src/Message/SmsMessageResult.php

```diff
@@ -48,6 +48,13 @@ class SmsMessageResult implements SmsMessageResultInterface {
   public $reports;
 
   /**
+   * The messages associated with this result.
```

Comment:
I'm a bit concerned about this. The initial model was for message results to be a property of an SMS message. So now with message results holding references to SMS messages, there's a bit of recursivity and data duplication here (especially with large number of recipients).
I'll look in more detail at this to figure out an optimal way of associating messages and message results.